src/divviProtocol/registerReferral.ts

```diff
@@ -0,0 +1,16 @@
+import { registryContractAbi } from 'src/divviProtocol/abi/Registry'
+import { REGISTRY_CONTRACT_ADDRESS } from 'src/divviProtocol/constants'
+import { SerializableTransactionRequest } from 'src/viem/preparedTransactionSerialization'
+import { TransactionRequest } from 'src/viem/prepareTransactions'
+import { decodeFunctionData } from 'viem'
+
+export function isRegistrationTransaction(tx: TransactionRequest | SerializableTransactionRequest) {
+  return (
+    tx.to === REGISTRY_CONTRACT_ADDRESS &&
+    tx.data &&
+    decodeFunctionData({
```

Comment:
This can throw, should we catch then? or is it handled by callers?

Reply:
i added a catch - i thought this would be unlikely to throw since we prepare the transaction internally, but indeed it's more correct to catch it here